<h1>Introduction/Business Problem</h1>

Predicting single family home foreclosure risk is an inexact science which in the past twenty years has largely depended on statistical analysis. To date, statistical analysis is heavily reliant on industry data, such as buyer lending information or static locations (near freeways, schools, etc). But from looking at social media data like what is found on FourSquare, there exists an opportunity to consider a change to the existing banking model that would make predicting foreclosure risk more dynamic and therefore accurate. If true, lending banks could save significant amounts of money from the thousands of automated home mortgage decisions they make, beyond what they are making with real estate solution products available through companies like CoreLogic or SiteXData. 

The potential model to a solution for this problem comes from what this capstone course has already accomplished, that is, analyzing Foursquare neighborhood venue data for the city of Toronto. In this  case, however, I will be applying the code to the city of Los Angeles. However, it is not just a copy. Multiple adjusted steps are needed for this to work, well beyond what has been completed.

<h1>Data</h1> 

The steps to solve the above business problem are the following:
<OL>
    <LI>The lat/lon gecoding information for the city of Los Angeles will need to be integrated with LA zipcode/neighborhood data that is scraped from the web.</LI>
    <LI>2018 foreclosure CVS file data from the city of Los Angeles will also need to be combined into the Los Angeles table in step 1.</LI>
    <LI>Foursquare will then be used to explore Los Angeles, coming up with venues for each neighborhood.</LI>
    <LI>Venues will then be organized by categories.</LI>
    <LI>Each venue category will include foreclosure frequency within a 1 mile radius. For example, the result will include the number of 2018 foreclosures for the category "Fast Food Restaurants."</LI>
    <LI>The data can then be analyzed for consistent foreclosure trends in each neighborhood related to venues.</LI>
    <LI>From step number 6, charts can be made to show how specific venues either increase, have no impact, or reduce foreclosure risk.</LI>
</OL>
<p>
An example of potentially conclusive venue data would be the following:</p>

<table>
    <tr><td>RESULT</td><td>EFFECT</td></tr>
    <tr><td>increased frequencies of Los Angeles parks</td><td>consistent lower than average foreclosure risk</td></tr>
    <tr><td>increased frequency of Los Angeles cafe's</td><td>normal range foreclosure risk</td></tr>
    <tr><td>increased frequency of Los Angeles fast food restaurants</td><td>consistent above average foreclosure risk</td></tr>
</table>

The next step will be to apply the above code and analysis to the cities of Boston, Seattle, Chicago and Houston, using FourSquare data, to see if the results for 2018 data are repeatable there. If the results are consistent, this study can lead to a deeper mathematical analyis as the next step in another project.

<h1>Methodology</h1>

<h2>Step 1: Loading 2018 LA foreclosure data, including geocodes, from a CSV file 100 records at at a time into a new DF</h2>

<h3>https://docs.google.com/spreadsheets/d/1J6PCrub6X1MuFwt6ZSxBx0Jv-McbnOuCQIZzKJdeK7I/edit#gid=393329626</h3>

In [4]:
#DEFINITIONS
import pandas as pd
#  https://docs.google.com/spreadsheets/d/16OMQGiVATyPgUCCxRfxw5vDKjmSQ5XIQwPQqgbop828/edit#gid=1171515980  (last version)
# entire 3100 rows  fc_data = pd.read_csv('https://docs.google.com/spreadsheets/d/16OMQGiVATyPgUCCxRfxw5vDKjmSQ5XIQwPQqgbop828/export?format=csv&gid=1171515980',
fc_data = pd.read_csv('https://docs.google.com/spreadsheets/d/1J6PCrub6X1MuFwt6ZSxBx0Jv-McbnOuCQIZzKJdeK7I/export?format=csv&gid=393329626',
                   # Set first column as rownames in data frame
                   header='infer', index_col=None, engine='python', error_bad_lines=False
                  )
fc_data.head(20)

# https://kanoki.org/2018/12/25/read-google-spreadsheet-data-into-pandas-dataframe/

APN RegisteredDate   PropertyType                    Address  \
0   6003016019     10/31/2018   Multi-Family             1217 W 60TH PL   
1   2151015040      11/1/2018  Single Family        20519 W HATTERAS ST   
2   5442030015      11/5/2018  Single Family            2801 N MARSH ST   
3   5015020012      1/17/2018   Multi-Family             2071 W 52ND ST   
4   7416002028       1/5/2018  Single Family          649 N FRIGATE AVE   
5   2162005011     11/10/2018  Single Family       5257 N NEWCASTLE AVE   
6   6006027004      11/6/2018  Single Family              500 E 59TH PL   
7   5460010052     11/21/2018  Single Family        2409 N YORKSHIRE DR   
8   2524021020       1/4/2018  Single Family          13279 W VAUGHN ST   
9   2425016027       2/2/2018  Single Family        3548 N MULTIVIEW DR   
10  2359022001      2/16/2018  Single Family      4758 N SUNNYSLOPE AVE   
11  4370022014      1/26/2018  Single Family  901 N STRADA VECCHIA ROAD   
12  4383006022      1/25/2018  Single Family          9847 W PORTOLA DR   
13  5065019024     10/24/2018   Multi-Family         5869 W DAUPHIN AVE   
14  2125007008       6/7/2018  Single Family        6951 N ETIWANDA AVE   
15  2261035019     12/14/2018  Single Family        15468 W MORRISON ST   
16  2217011057     11/12/2018   Multi-Family         14235 W VANOWEN ST   
17  4403011006     12/12/2018  Single Family       213 N TIGERTAIL ROAD   
18  7414015018       1/1/2018  Single Family            1021 W LOWEN ST   
19  4321015010       1/1/2018  Single Family         2038 S MALCOLM AVE   

           City State    Zip   Latitude   Longitude  
0   LOS ANGELES    CA  90044  33.985025 -118.296540  
1   LOS ANGELES    CA  91367  34.175498 -118.579552  
2   LOS ANGELES    CA  90039  34.104301 -118.248531  
3   LOS ANGELES    CA  90062  33.995044 -118.316025  
4   LOS ANGELES    CA  90744  33.777863 -118.278247  
5        ENCINO    CA  91316  34.166353 -118.524347  
6   LOS ANGELES    CA  90003  33.986083 -118.266182  
7   LOS ANGELES    CA  90065  34.110697 -118.230535  
8   LOS ANGELES    CA  91340  34.281704 -118.422398  
9   LOS ANGELES    CA  90068  34.130287 -118.360437  
10  LOS ANGELES    CA  91423  34.157384 -118.426876  
11  LOS ANGELES    CA  90077  34.089718 -118.443840  
12  LOS ANGELES    CA  90210  34.110322 -118.433459  
13  LOS ANGELES    CA  90034  27.871628  -82.498600  
14  LOS ANGELES    CA  91335  34.197043 -118.531672  
15  LOS ANGELES    CA  91403  34.160296 -118.471627  
16     VAN NUYS    CA  91405  34.193929 -118.443275  
17  LOS ANGELES    CA  90049  34.061779 -118.479050  
18  LOS ANGELES    CA  90744  33.795173 -118.275929  
19  LOS ANGELES    CA  90025  34.046825 -118.431553

<h2>Step 2: Exploring and Clustering Foreclosed Addresses (2018) in Los Angeles</h2>

<h3>First, get lat/lon of Los Angeles</h3>

In [7]:
# Use geopy library to get the latitude and longitude values of Los Angeles
from geopy.geocoders import Nominatim 
#Nominatim is a tool to search OSM (open street map) data by name and address and to generate synthetic addresses of OSM points (reverse geocoding).
import matplotlib.cm as cm
import matplotlib.colors as colors
from sklearn.cluster import KMeans
!conda install -c conda-forge folium=0.5.0 --yes
import folium # folium is mapping tool
print('Libraries imported.')

Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/DSX-Python35

  added / updated specs: 
    - folium=0.5.0


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-1.0.2r             |       h14c3975_0         3.1 MB  conda-forge
    folium-0.5.0               |             py_0          45 KB  conda-forge
    vincent-0.4.4              |             py_1          28 KB  conda-forge
    altair-2.2.2               |           py35_1         462 KB  conda-forge
    ca-certificates-2019.3.9   |       hecc5488_0         146 KB  conda-forge
    branca-0.3.1               |             py_0          25 KB  conda-forge
    certifi-2018.8.24          |        py35_1001         139 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         4.0 MB

The following NEW packages will

In [8]:
address = 'Los Angeles, California'
geolocator = Nominatim(user_agent="Los Angeles")
location = geolocator.geocode(address)
latitude_LA = location.latitude
longitude_LA = location.longitude
print('The geograpical coordinate of Los Angeles are {}, {}.'.format(latitude_LA, longitude_LA))

The geograpical coordinate of Los Angeles are 34.0536909, -118.2427666.


<h3>Show Map of Los Angeles</h3>

In [9]:
map_LA = folium.Map(location=[latitude_LA, longitude_LA], zoom_start=10)

# add markers to map
for lat, lng, address, zipcode in zip(fc_data['Latitude'], fc_data['Longitude'], fc_data['Address'], fc_data['Zip']):
    label = '{}, {}'.format(zipcode, address)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_LA)  
    
map_LA

<h3>Define Foursquare Credentials and Version</h3>

In [10]:
CLIENT_ID = "RBX2RESNE1HWSSGX020MOSI4CTL4SIP4QQPZFGJ2LF4EBOMX"  #Foursquare ID
CLIENT_SECRET = "YBZIPEL5GKJX2T13SPD5TDM4ZFXRV51SO2WDNKEGUTSMIWIU"
# was CLIENT_SECRET = "H2LKEFPFFGH4L3WLNDFLPVF0O11S55UMOAIUEX0S4MHH3GAV"  #Foursquare Secret

VERSION = "20180605"  #API Version

print("Your Credentials:")
print("CLIENT ID:" + CLIENT_ID)
print("CLIENT SECRET" + CLIENT_SECRET)

Your Credentials:
CLIENT ID:RBX2RESNE1HWSSGX020MOSI4CTL4SIP4QQPZFGJ2LF4EBOMX
CLIENT SECRETYBZIPEL5GKJX2T13SPD5TDM4ZFXRV51SO2WDNKEGUTSMIWIU


In [11]:
# defining radius and limit of venues to get
radius=500
LIMIT=100

<h3>Explore the first zip in the Dataframe.</h3>

In [12]:
# Get the first address zip
fc_data.loc[0, 'Zip']

90044

<h3>Get the address latitude and longitude values.</h3>

In [13]:

address_latitude = fc_data.loc[0, 'Latitude'] # address latitude value
address_longitude = fc_data.loc[0, 'Longitude'] # address longitude value
address_zip = fc_data.loc[0, 'Zip'] # address Zip

print('Latitude and longitude values of {} are {}, {}.'.format(address_zip, 
                                                               address_latitude, 
                                                               address_longitude))

Latitude and longitude values of 90044 are 33.985025, -118.29654.


<h3>The Top 100 venues that are in zip 90044 within a radius of 500 meters.</h3>

In [14]:
#First, let's create the GET request URL.
# radius = 500
# LIMIT = 100

url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, 
    CLIENT_SECRET, 
    VERSION, 
    address_latitude,
    address_longitude, 
    radius, 
    LIMIT)
    
url  #display URL

'https://api.foursquare.com/v2/venues/explore?&client_id=RBX2RESNE1HWSSGX020MOSI4CTL4SIP4QQPZFGJ2LF4EBOMX&client_secret=YBZIPEL5GKJX2T13SPD5TDM4ZFXRV51SO2WDNKEGUTSMIWIU&v=20180605&ll=33.985025,-118.29654&radius=500&limit=100'

In [15]:
#Send the GET Request and examine the results
import requests

results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5cf346694c1f6753b8283024'},
 'response': {'groups': [{'items': [{'reasons': {'count': 0,
       'items': [{'reasonName': 'globalInteractionReason',
         'summary': 'This spot is popular',
         'type': 'general'}]},
      'referralId': 'e-0-4fcaab5ee4b0d8a39149da92-0',
      'venue': {'categories': [{'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/fishandchips_',
          'suffix': '.png'},
         'id': '4edd64a0c7ddd24ca188df1a',
         'name': 'Fish & Chips Shop',
         'pluralName': 'Fish & Chips Shops',
         'primary': True,
         'shortName': 'Fish & Chips'}],
       'id': '4fcaab5ee4b0d8a39149da92',
       'location': {'address': '5950 S Normandie Ave',
        'cc': 'US',
        'city': 'Los Angeles',
        'country': 'United States',
        'crossStreet': 'Normandie & 59th',
        'distance': 369,
        'formattedAddress': ['5950 S Normandie Ave (Normandie & 59th)',
         'Los Angeles, CA 9004

<h3>Get the category of the venue</h3>

In [16]:
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

<h3>Clean the json and structure it into a pandas dataframe.</h3>

In [17]:
import json # library to handle JSON files
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

venues = results['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues =nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

nearby_venues.head()

name                 categories        lat         lng
0   east west fish market          Fish & Chips Shop  33.986600 -118.300067
1  Don Lenchos Restaurant                 Restaurant  33.983687 -118.300546
2     Happy Laundry 24hrs                 Laundromat  33.986883 -118.300362
3               Amapulapa  Latin American Restaurant  33.987215 -118.300351
4        Normandie / Gage                   Bus Stop  33.982140 -118.300170

In [18]:
# And how many venues were returned by Foursquare?
print('{} venues were returned by Foursquare.'.format(nearby_venues.shape[0]))

5 venues were returned by Foursquare.


<h3>Explore Neighborhoods in Los Angeles 100 rows at a time. Repeat same process for all 2018 foreclosured addresses in Los Angeles.</h3>

In [19]:
def getNearbyVenues(address, city, zipcode, latitude, longitude, radius=500):
    
    venues_list=[]
    for addr, cty, zcode, lat, lng in zip(address, city, zipcode, latitude, longitude):
        print(addr)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            addr,
            zcode,
            lat, 
            lng,
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['FC Address', 
                  'FC Zip',
                  'FC Latitude',
                  'FC Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

<h3>Create code to run the above function on each foreclosed address and create a new dataframe called LA_venues.</h3>

In [20]:
LA_venues = getNearbyVenues(address=fc_data['Address'],
                                   city=fc_data['City'],
                                   zipcode=fc_data['Zip'],
                                   latitude=fc_data['Latitude'],
                                   longitude=fc_data['Longitude']
                                  )
# LA_venues.head(25)

1217 W 60TH PL
20519 W HATTERAS ST
2801 N MARSH ST
2071 W 52ND ST
649 N FRIGATE AVE
5257 N NEWCASTLE AVE
500 E 59TH PL
2409 N YORKSHIRE DR
13279 W VAUGHN ST
3548 N MULTIVIEW DR
4758 N SUNNYSLOPE AVE
901 N STRADA VECCHIA ROAD
9847 W PORTOLA DR
5869 W DAUPHIN AVE
6951 N ETIWANDA AVE
15468 W MORRISON ST
14235 W VANOWEN ST
213 N TIGERTAIL ROAD
1021 W LOWEN ST
2038 S MALCOLM AVE
15935 W COMMUNITY ST
6901 W CAHUENGA PARK TR
1027 S CENTRE ST
23222 W AETNA ST
13640 W GARBER ST
5211 N DARRO ROAD
20321 W VIA SANSOVINO
1916 N HANCOCK ST
929 S MUIRFIELD RD
10601 W WILSHIRE BLVD 1-72
7217 S DENKER AVE
12383 W COVELLO ST
7014 N COLBATH AVE
3749 N GLENALBYN DR
1334 E 57TH ST
11622 W KESWICK ST
9470 S HOBART BLVD
2568 S MILITARY AVE
635 S BYNNER DR
1240 N RAVENNA AVE
1871 W 25TH ST
2651 N ABERDEEN AVE
7129 W HILLROSE ST
19501 W ROMAR ST
1445  YOSEMITE DR
14325 W TORI CT
4868 N ADELE CT
1916 W WHITMORE AVE
11846 S ORCHARD AVE
727 N ALTA VISTA BLVD
11221 N AMBOY AVE
17169 W STARE ST
6242 W TEMPLE HILL D

<h3>Check the size of the resulting dataframe</h3>

In [21]:
print(LA_venues.shape)
LA_venues.head()

(952, 8)


FC Address  FC Zip  FC Latitude  FC Longitude                   Venue  \
0  1217 W 60TH PL   90044    33.985025    -118.29654   east west fish market   
1  1217 W 60TH PL   90044    33.985025    -118.29654  Don Lenchos Restaurant   
2  1217 W 60TH PL   90044    33.985025    -118.29654     Happy Laundry 24hrs   
3  1217 W 60TH PL   90044    33.985025    -118.29654               Amapulapa   
4  1217 W 60TH PL   90044    33.985025    -118.29654        Normandie / Gage   

   Venue Latitude  Venue Longitude             Venue Category  
0       33.986600      -118.300067          Fish & Chips Shop  
1       33.983687      -118.300546                 Restaurant  
2       33.986883      -118.300362                 Laundromat  
3       33.987215      -118.300351  Latin American Restaurant  
4       33.982140      -118.300170                   Bus Stop

<h3>Check how many venues were returned for each LA Foreclosed (FC) Address</h3>

In [22]:
LA_venues.groupby('FC Address').count()

FC Zip  FC Latitude  FC Longitude  Venue  \
FC Address                                                                
1000 E 74TH ST                      4            4             4      4   
10000 N OWENSMOUTH AVE             13           13            13     13   
1021 W LOWEN ST                     4            4             4      4   
1027 S CENTRE ST                   23           23            23     23   
10650 W CHIQUITA ST                12           12            12     12   
1077 S RIMPAU BLVD                  9            9             9      9   
11163 N TELFAIR AVE                 3            3             3      3   
11221 N AMBOY AVE                   6            6             6      6   
11407 N SANTINI LANE                3            3             3      3   
11622 W KESWICK ST                 13           13            13     13   
11846 S ORCHARD AVE                 2            2             2      2   
12006 N HIGHWATER ROAD              1            1             1      1   
1217 W 60TH PL                      5            5             5      5   
1224 E 57TH ST                     18           18            18     18   
12383 W COVELLO ST                  1            1             1      1   
1240 N RAVENNA AVE                  7            7             7      7   
1270 S MEADOWBROOK AVE             20           20            20     20   
12818 W FRIAR ST                   22           22            22     22   
1301 W 84TH ST                      9            9             9      9   
13279 W VAUGHN ST                   4            4             4      4   
1334 E 57TH ST                      8            8             8      8   
13375 N PHILLIPPI AVE               1            1             1      1   
13463 W HUBBARD ST                 12           12            12     12   
13640 W GARBER ST                   2            2             2      2   
1405 E WARREN ST                   15           15            15     15   
14235 W VANOWEN ST                  8            8             8      8   
14325 W TORI CT                     2            2             2      2   
14333 W VAN NUYS BLVD              10           10            10     10   
14348 W TIARA ST                    5            5             5      5   
1445  YOSEMITE DR                   2            2             2      2   
...                               ...          ...           ...    ...   
4868 N ADELE CT                     1            1             1      1   
500 E 59TH PL                       5            5             5      5   
5211 N DARRO ROAD                   8            8             8      8   
5243 N NORTH MAYWOOD AVE           33           33            33     33   
5257 N NEWCASTLE AVE               20           20            20     20   
550 N PASEO MIRAMAR                 9            9             9      9   
5869 W DAUPHIN AVE                  2            2             2      2   
620 N NEPTUNE AVE                  10           10            10     10   
6242 W TEMPLE HILL DR              11           11            11     11   
649 N FRIGATE AVE                  11           11            11     11   
6701 N SALE AVE                     5            5             5      5   
6901 W CAHUENGA PARK TR             4            4             4      4   
6951 N ETIWANDA AVE                29           29            29     29   
7014 N COLBATH AVE                  4            4             4      4   
7129 W HILLROSE ST                  2            2             2      2   
7217 S DENKER AVE                   7            7             7      7   
7241 W WOODROW WILSON DR            2            2             2      2   
727 N ALTA VISTA BLVD              84           84            84     84   
806 N MANSFIELD AVE                44           44            44     44   
8278 N GLENCREST DR                 2            2             2      2   
831 E 94TH ST                       4            4      

<h3>Find out how many unique categories can be curated from all the returned LA venues</h3>

In [23]:
print('There are {} unique categories.'.format(len(LA_venues['Venue Category'].unique())))

There are 217 unique categories.


<h3>Analyze Each LA Foreclosed Address</h3>

In [24]:
# one hot encoding
LA_onehot = pd.get_dummies(LA_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
LA_onehot['FC Address'] = LA_venues['FC Address'] 

# move neighborhood column to the first column
fixed_columns = [LA_onehot.columns[-1]] + list(LA_onehot.columns[:-1])
LA_onehot = LA_onehot[fixed_columns]

LA_onehot.head()

FC Address  ATM  Airport Terminal  American Restaurant  \
0  1217 W 60TH PL    0                 0                    0   
1  1217 W 60TH PL    0                 0                    0   
2  1217 W 60TH PL    0                 0                    0   
3  1217 W 60TH PL    0                 0                    0   
4  1217 W 60TH PL    0                 0                    0   

   Argentinian Restaurant  Art Gallery  Arts & Crafts Store  Asian Restaurant  \
0                       0            0                    0                 0   
1                       0            0                    0                 0   
2                       0            0                    0                 0   
3                       0            0                    0                 0   
4                       0            0                    0                 0   

   Athletics & Sports  Auto Workshop     ...       Video Game Store  \
0                   0              0     ...                      0   
1                   0              0     ...                      0   
2                   0              0     ...                      0   
3                   0              0     ...                      0   
4                   0              0     ...                      0   

   Video Store  Vietnamese Restaurant  Warehouse Store  Weight Loss Center  \
0            0                      0                0                   0   
1            0                      0                0                   0   
2            0                      0                0                   0   
3            0                      0                0                   0   
4            0                      0                0                   0   

   Wine Bar  Wine Shop  Wings Joint  Women's Store  Yoga Studio  
0         0          0            0              0            0  
1         0          0            0              0            0  
2         0          0            0              0            0  
3         0          0            0              0            0  
4         0          0            0              0            0  

[5 rows x 218 columns]

<h3>What is the new dataframe size?</h3>

In [25]:
LA_onehot.shape

(952, 218)

<h3>Group rows by LA FC Address by taking the mean of the frequency of occurrence of each category.</h3>

In [26]:
LA_grouped = LA_onehot.groupby('FC Address').mean().reset_index()
LA_grouped

FC Address       ATM  Airport Terminal  \
0                  1000 E 74TH ST  0.000000          0.000000   
1          10000 N OWENSMOUTH AVE  0.000000          0.000000   
2                 1021 W LOWEN ST  0.000000          0.000000   
3                1027 S CENTRE ST  0.000000          0.000000   
4             10650 W CHIQUITA ST  0.000000          0.000000   
5              1077 S RIMPAU BLVD  0.000000          0.111111   
6             11163 N TELFAIR AVE  0.000000          0.000000   
7               11221 N AMBOY AVE  0.000000          0.000000   
8            11407 N SANTINI LANE  0.000000          0.000000   
9              11622 W KESWICK ST  0.000000          0.000000   
10            11846 S ORCHARD AVE  0.000000          0.000000   
11         12006 N HIGHWATER ROAD  0.000000          0.000000   
12                 1217 W 60TH PL  0.000000          0.000000   
13                 1224 E 57TH ST  0.000000          0.000000   
14             12383 W COVELLO ST  0.000000          0.000000   
15             1240 N RAVENNA AVE  0.000000          0.000000   
16         1270 S MEADOWBROOK AVE  0.000000          0.000000   
17               12818 W FRIAR ST  0.000000          0.000000   
18                 1301 W 84TH ST  0.000000          0.000000   
19              13279 W VAUGHN ST  0.000000          0.000000   
20                 1334 E 57TH ST  0.000000          0.000000   
21          13375 N PHILLIPPI AVE  0.000000          0.000000   
22             13463 W HUBBARD ST  0.000000          0.000000   
23              13640 W GARBER ST  0.000000          0.000000   
24               1405 E WARREN ST  0.000000          0.000000   
25             14235 W VANOWEN ST  0.000000          0.000000   
26                14325 W TORI CT  0.000000          0.000000   
27          14333 W VAN NUYS BLVD  0.000000          0.000000   
28               14348 W TIARA ST  0.000000          0.000000   
29              1445  YOSEMITE DR  0.000000          0.000000   
..                            ...       ...               ...   
65                4868 N ADELE CT  0.000000          0.000000   
66                  500 E 59TH PL  0.000000          0.000000   
67              5211 N DARRO ROAD  0.000000          0.000000   
68       5243 N NORTH MAYWOOD AVE  0.000000          0.000000   
69           5257 N NEWCASTLE AVE  0.000000          0.000000   
70            550 N PASEO MIRAMAR  0.000000          0.000000   
71             5869 W DAUPHIN AVE  0.000000          0.000000   
72              620 N NEPTUNE AVE  0.000000          0.000000   
73          6242 W TEMPLE HILL DR  0.000000          0.000000   
74              649 N FRIGATE AVE  0.000000          0.000000   
75                6701 N SALE AVE  0.000000          0.000000   
76        6901 W CAHUENGA PARK TR  0.000000          0.000000   
77            6951 N ETIWANDA AVE  0.000000          0.000000   
78             7014 N COLBATH AVE  0.000000          0.000000   
79             7129 W HILLROSE ST  0.000000          0.000000   
80              7217 S DENKER AVE  0.000000          0.000000   
81       7241 W WOODROW WILSON DR  0.000000          0.000000   
82          727 N ALTA VISTA BLVD  0.011905          0.000000   
83            806 N MANSFIELD AVE  0.000000          0.000000   
84            8278 N GLENCREST DR  0.000000          0.000000   
85                  831 E 94TH ST  0.000000          0.000000   
86         8620 S BELFORD AVE A-F  0.000000          0.000000   
87      901 N STRADA VECCHIA ROAD  0.000000          0.000000   
88  9113 #3 S BARING CROSS ST 1-3  0.000000          0.000000   
89                  917 W 23RD ST  0.000000          0.000000   
90          9211 W NIGHTINGALE DR  0.000000          0.000000   
91             929 S MUIRFIELD RD  0.000000          0.250000   
92             9470 S HOBART BLVD  0.000000          0.000000   
93         9503 N HAYVENHURST AVE  0.000000          0.000000   
94              9847 W PORTOLA DR  0.000000          0.000

<h3>Confirm the new size</h3>

In [27]:
LA_grouped.shape

(95, 218)

<h3>Print each LA FC address along with the top 5 most common LA venues</h3>

In [28]:
num_top_venues = 5

for address in LA_grouped['FC Address']:
    print("----"+address+"----")
    temp = LA_grouped[LA_grouped['FC Address'] == address].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----1000 E 74TH ST----
                        venue  freq
0  Construction & Landscaping  0.25
1          Mexican Restaurant  0.25
2          Seafood Restaurant  0.25
3                        Bank  0.25
4                         ATM  0.00


----10000 N OWENSMOUTH AVE----
                           venue  freq
0            Japanese Restaurant  0.08
1  Vegetarian / Vegan Restaurant  0.08
2                     Smoke Shop  0.08
3                    Coffee Shop  0.08
4             Persian Restaurant  0.08


----1021 W LOWEN ST----
                  venue  freq
0        Baseball Field  0.25
1   Fried Chicken Joint  0.25
2  Gym / Fitness Center  0.25
3            Hobby Shop  0.25
4                   ATM  0.00


----1027 S CENTRE ST----
              venue  freq
0   Thai Restaurant  0.09
1  Sushi Restaurant  0.04
2               Bar  0.04
3        Sports Bar  0.04
4        Steakhouse  0.04


----10650 W CHIQUITA ST----
                venue  freq
0  Italian Restaurant  0.17
1                Pa

<h3>Put above into a pandas dataframe</h3>

In [35]:
# First write a function to sort the venues in descending order.

def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    return row_categories_sorted.index.values[0:num_top_venues]

In [36]:
num_top_venues = 10
import numpy as np # library to handle data in a vectorized manner

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['FC Address']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
fcaddress_venues_sorted = pd.DataFrame(columns=columns)
fcaddress_venues_sorted['FC Address'] = LA_grouped['FC Address']

for ind in np.arange(LA_grouped.shape[0]):
    fcaddress_venues_sorted.iloc[ind, 1:] = return_most_common_venues(LA_grouped.iloc[ind, :], num_top_venues)

fcaddress_venues_sorted.head()

FC Address 1st Most Common Venue 2nd Most Common Venue  \
0          1000 E 74TH ST                  Bank    Seafood Restaurant   
1  10000 N OWENSMOUTH AVE         Hot Dog Joint   Japanese Restaurant   
2         1021 W LOWEN ST  Gym / Fitness Center            Hobby Shop   
3        1027 S CENTRE ST       Thai Restaurant                 Hotel   
4     10650 W CHIQUITA ST            Food Truck    Italian Restaurant   

  3rd Most Common Venue       4th Most Common Venue 5th Most Common Venue  \
0    Mexican Restaurant  Construction & Landscaping      Football Stadium   
1    Salon / Barbershop                 Coffee Shop         Train Station   
2        Baseball Field         Fried Chicken Joint             Dive Shop   
3                   Bar                 Pizza Place   Rental Car Location   
4                  Park              Discount Store          Dance Studio   

  6th Most Common Venue 7th Most Common Venue          8th Most Common Venue  \
0            Food Truck            Food Stand                     Food Court   
1  Fast Food Restaurant    Persian Restaurant  Vegetarian / Vegan Restaurant   
2  Fast Food Restaurant           Frame Store               Football Stadium   
3        Clothing Store             Rock Club                 Sandwich Place   
4        Breakfast Spot                   Gym                            Bar   

  9th Most Common Venue 10th Most Common Venue  
0                  Food            Flower Shop  
1            Smoke Shop            Music Venue  
2            Food Truck             Food Stand  
3                  Café     Seafood Restaurant  
4      Ramen Restaurant             Donut Shop

<h3>Cluster FCAddresses--Run k-means to cluster the FCAdresses into 5 clusters.</h3>

In [37]:
# set number of clusters
kclusters = 5

LA_grouped_clustering = LA_grouped.drop('FC Address', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(LA_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_

array([1, 4, 4, 4, 4, 4, 4, 4, 2, 1, 4, 4, 4, 4, 4, 1, 4, 4, 4, 1, 4, 3, 4,
       4, 1, 1, 4, 4, 1, 4, 4, 1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 2, 4, 4, 4,
       2, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 1, 4, 4, 4, 0, 4, 4, 4,
       4, 4, 4, 1, 4, 4, 4, 2, 4, 1, 4, 4, 2, 4, 4, 1, 4, 4, 4, 4, 4, 4, 4,
       4, 4, 2], dtype=int32)

In [39]:
# create a new dataframe that includes the cluster as well as the top 10 venues for each neighborhood.
# add clustering labels
fcaddress_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)
LA_merged = fc_data

# merge LA_grouped with LA_data to add latitude/longitude for each address
LA_merged = LA_merged.join(fcaddress_venues_sorted.set_index('FC Address'), on='Address')
LA_merged.head(20) # check the last columns!

APN RegisteredDate   PropertyType                    Address  \
0   6003016019     10/31/2018   Multi-Family             1217 W 60TH PL   
1   2151015040      11/1/2018  Single Family        20519 W HATTERAS ST   
2   5442030015      11/5/2018  Single Family            2801 N MARSH ST   
3   5015020012      1/17/2018   Multi-Family             2071 W 52ND ST   
4   7416002028       1/5/2018  Single Family          649 N FRIGATE AVE   
5   2162005011     11/10/2018  Single Family       5257 N NEWCASTLE AVE   
6   6006027004      11/6/2018  Single Family              500 E 59TH PL   
7   5460010052     11/21/2018  Single Family        2409 N YORKSHIRE DR   
8   2524021020       1/4/2018  Single Family          13279 W VAUGHN ST   
9   2425016027       2/2/2018  Single Family        3548 N MULTIVIEW DR   
10  2359022001      2/16/2018  Single Family      4758 N SUNNYSLOPE AVE   
11  4370022014      1/26/2018  Single Family  901 N STRADA VECCHIA ROAD   
12  4383006022      1/25/2018  Single Family          9847 W PORTOLA DR   
13  5065019024     10/24/2018   Multi-Family         5869 W DAUPHIN AVE   
14  2125007008       6/7/2018  Single Family        6951 N ETIWANDA AVE   
15  2261035019     12/14/2018  Single Family        15468 W MORRISON ST   
16  2217011057     11/12/2018   Multi-Family         14235 W VANOWEN ST   
17  4403011006     12/12/2018  Single Family       213 N TIGERTAIL ROAD   
18  7414015018       1/1/2018  Single Family            1021 W LOWEN ST   
19  4321015010       1/1/2018  Single Family         2038 S MALCOLM AVE   

           City State    Zip   Latitude   Longitude  Cluster Labels  \
0   LOS ANGELES    CA  90044  33.985025 -118.296540             4.0   
1   LOS ANGELES    CA  91367  34.175498 -118.579552             4.0   
2   LOS ANGELES    CA  90039  34.104301 -118.248531             4.0   
3   LOS ANGELES    CA  90062  33.995044 -118.316025             4.0   
4   LOS ANGELES    CA  90744  33.777863 -118.278247             4.0   
5        ENCINO    CA  91316  34.166353 -118.524347             4.0   
6   LOS ANGELES    CA  90003  33.986083 -118.266182             4.0   
7   LOS ANGELES    CA  90065  34.110697 -118.230535             4.0   
8   LOS ANGELES    CA  91340  34.281704 -118.422398             1.0   
9   LOS ANGELES    CA  90068  34.130287 -118.360437             4.0   
10  LOS ANGELES    CA  91423  34.157384 -118.426876             4.0   
11  LOS ANGELES    CA  90077  34.089718 -118.443840             4.0   
12  LOS ANGELES    CA  90210  34.110322 -118.433459             2.0   
13  LOS ANGELES    CA  90034  27.871628  -82.498600             4.0   
14  LOS ANGELES    CA  91335  34.197043 -118.531672             4.0   
15  LOS ANGELES    CA  91403  34.160296 -118.471627             1.0   
16     VAN NUYS    CA  91405  34.193929 -118.443275             1.0   
17  LOS ANGELES    CA  90049  34.061779 -118.479050             4.0   
18  LOS ANGELES    CA  90744  33.795173 -118.275929             4.0   
19  LOS ANGELES    CA  90025  34.046825 -118.431553             4.0   

       1st Most Common Venue  2nd Most Common Venue  \
0                   Bus Stop             Restaurant   
1                       Pool       Business Service   
2     Furniture / Home Store             Food Truck   
3                       Park               Wine Bar   
4                 Donut Shop           Burger Joint   
5                Video Store            Pizza Place   
6                 Taco Place         Clothing Store   
7           Business Service                   Café   
8                 Taco Place      Convenience Store   
9                Yoga Studio   Marijuana Dispensary   
10            Cosmetics Shop                    ATM   
11                Hotel Pool              Hotel Bar   
12                     Trail            Yoga Studio   
13        Italian Restaurant         Cosmetics Shop   
14                  Pharmacy   Fast Food Restaurant   
15             Moving Target  General Entertainment   
16      

<h3>Visualize the resulting clusters</h3>

In [40]:
# create map
map_clusters = folium.Map(location=[latitude_LA, longitude_LA], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(LA_merged['Latitude'], LA_merged['Longitude'], LA_merged['Address'], LA_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        #color=rainbow[cluster-1],
        fill=True,
        # fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)

folium.map.LayerControl('topright', collapsed=False).add_to(map_clusters)
map_clusters

<h3>Examine each cluster and determine the discriminating venue categories that distinguish each cluster.</h3>

<h3>Cluster 1</h3>

In [41]:
LA_merged.loc[LA_merged['Cluster Labels'] == 0, LA_merged.columns[[3] + list(range(5, LA_merged.shape[1]))]]

Address State    Zip   Latitude   Longitude  Cluster Labels  \
46  4868 N ADELE CT    CA  91364  34.159318 -118.572731             0.0   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
46            Smoke Shop           Yoga Studio        Farmers Market   

   4th Most Common Venue 5th Most Common Venue 6th Most Common Venue  \
46      Football Stadium            Food Truck            Food Stand   

   7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
46            Food Court                  Food           Flower Shop   

   10th Most Common Venue  
46            Flea Market

<h3>Cluster 2</h3>

In [97]:
LA_merged.loc[LA_merged['Cluster Labels'] == 1, LA_merged.columns[[3] + list(range(5, LA_merged.shape[1]))]]

Address State    Zip   Latitude   Longitude  Cluster Labels  \
8     13279 W VAUGHN ST    CA  91340  34.281704 -118.422398             1.0   
15  15468 W MORRISON ST    CA  91403  34.160296 -118.471627             1.0   
16   14235 W VANOWEN ST    CA  91405  34.193929 -118.443275             1.0   
27    1916 N HANCOCK ST    CA  90031  34.066688 -118.208930             1.0   
32   7014 N COLBATH AVE    CA  91405  34.198098 -118.436712             1.0   
33  3749 N GLENALBYN DR    CA  90065  34.093187 -118.216053             1.0   
35   11622 W KESWICK ST    CA  91605  34.210274 -118.385325             1.0   
39   1240 N RAVENNA AVE    CA  90744  33.787835 -118.268817             1.0   
58       4235 W 59TH ST    CA  90043  33.987090 -118.348847             1.0   
71       1000 E 74TH ST    CA  90001  33.973003 -118.257629             1.0   
77     14348 W TIARA ST    CA  91401  34.178458 -118.446019             1.0   
86     1405 E WARREN ST    CA  90033  34.051221 -118.220076             1.0   
92  8278 N GLENCREST DR    CA  91352  34.221362 -118.356804             1.0   
93    620 N NEPTUNE AVE    CA  90744  33.777386 -118.268768             1.0   

   1st Most Common Venue            2nd Most Common Venue  \
8             Taco Place                Convenience Store   
15         Moving Target            General Entertainment   
16              Dive Bar                Convenience Store   
27    Mexican Restaurant                     Burger Joint   
32    Mexican Restaurant                Convenience Store   
33          Liquor Store               Mexican Restaurant   
35   Rental Car Location                     Liquor Store   
39          Dessert Shop                             Bank   
58    Seafood Restaurant  Southern / Soul Food Restaurant   
71                  Bank               Seafood Restaurant   
77   Rental Car Location                Convenience Store   
86    Mexican Restaurant                     Intersection   
92    Mexican Restaurant                         Gym Pool   
93    Mexican Restaurant                   Discount Store   

     3rd Most Common Venue       4th Most Common Venue 5th Most Common Venue  \
8       Athletics & Sports          Mexican Restaurant           Yoga Studio   
15               BBQ Joint        Fast Food Restaurant     French Restaurant   
16  Furniture / Home Store              Breakfast Spot    Mexican Restaurant   
27                    Park                  Food Truck          Optical Shop   
32             Video Store                 Yoga Studio        Farmers Market   
33              Smoke Shop                      Lawyer        Farmers Market   
35      Mexican Restaurant                Burger Joint  Gym / Fitness Center   
39      Mexican Restaurant          Seafood Restaurant          Burger Joint   
58          Cosmetics Shop           Convenience Store           Video Store   
71      Mexican Restaurant  Construction & Landscaping      Football Stadium   
77               BBQ Joint                Skating Rink           Yoga Studio   
86                  Bakery                        Park        Sandwich Place   
92             Yoga Studio        Fast Food Restaurant           Frame Store   
93             Video Store               Grocery Store           Coffee Shop   

   6th Most Common Venue      7th Most Common Venue 8th Most Common Venue  \
8   Fast Food Restaurant           Football Stadium            Food Truck   
15           Frame Store           Football Stadium            Food Truck   
16                  Bank                  BBQ Joint              Gym Pool   
27                  Lake           Football Stadium            Food Stand   
32      Football Stadium                 Food Truck            Food Stand   
33      Football Stadium                 Food Truck            Food Stand   
35     Convenience Store              Auto Workshop      Sushi Restaurant   
39          Liquor Store                Pizza Place               Dog Run   
58    Mexican Restau

<h3>Cluster 3</h3>

In [98]:
LA_merged.loc[LA_merged['Cluster Labels'] == 2, LA_merged.columns[[3] + list(range(5, LA_merged.shape[1]))]]

Address State    Zip   Latitude   Longitude  \
12         9847 W PORTOLA DR    CA  90210  34.110322 -118.433459   
21   6901 W CAHUENGA PARK TR    CA  90068  34.123224 -118.345339   
47       1916 W WHITMORE AVE    CA  90039  34.095644 -118.251188   
70      11407 N SANTINI LANE    CA  91326  34.278566 -118.587486   
79         20628 W COMO LANE    CA  91326  34.277867 -118.582721   
87  7241 W WOODROW WILSON DR    CA  90068  34.124095 -118.350240   

    Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
12             2.0                 Trail           Yoga Studio   
21             2.0             Rock Club             Piano Bar   
47             2.0                 Trail           Music Venue   
70             2.0                 Trail                  Pool   
79             2.0                  Pool                 Trail   
87             2.0                 Trail             Rock Club   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
12   Fried Chicken Joint           Frame Store      Football Stadium   
21  Marijuana Dispensary                 Trail           Yoga Studio   
47          Liquor Store            Food Truck        Pilates Studio   
70           Yoga Studio      Football Stadium            Food Truck   
79        Scenic Lookout            Food Truck            Food Stand   
87           Yoga Studio      Football Stadium            Food Truck   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
12            Food Truck            Food Stand            Food Court   
21           Fabric Shop            Food Truck            Food Stand   
47      Basketball Court            Food Stand            Food Court   
70            Food Stand            Food Court                  Food   
79            Food Court                  Food           Flower Shop   
87            Food Stand            Food Court                  Food   

   9th Most Common Venue 10th Most Common Venue  
12                  Food            Flower Shop  
21            Food Court                   Food  
47                  Food            Flower Shop  
70           Flower Shop            Flea Market  
79           Flea Market      Fish & Chips Shop  
87           Flower Shop            Flea Market

<h3>Cluster 4</h3>

In [99]:
LA_merged.loc[LA_merged['Cluster Labels'] == 3, LA_merged.columns[[3] + list(range(5, LA_merged.shape[1]))]]

Address State    Zip   Latitude   Longitude  Cluster Labels  \
61  13375 N PHILLIPPI AVE    CA  91342  34.315109 -118.449756             3.0   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
61          Home Service           Yoga Studio           Fabric Shop   

   4th Most Common Venue 5th Most Common Venue 6th Most Common Venue  \
61      Football Stadium            Food Truck            Food Stand   

   7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
61            Food Court                  Food           Flower Shop   

   10th Most Common Venue  
61            Flea Market

<h3>Cluster 5</h3>

In [100]:
LA_merged.loc[LA_merged['Cluster Labels'] == 4, LA_merged.columns[[3] + list(range(5, LA_merged.shape[1]))]]

Address State    Zip   Latitude   Longitude  \
0                  1217 W 60TH PL    CA  90044  33.985025 -118.296540   
1             20519 W HATTERAS ST    CA  91367  34.175498 -118.579552   
2                 2801 N MARSH ST    CA  90039  34.104301 -118.248531   
3                  2071 W 52ND ST    CA  90062  33.995044 -118.316025   
4               649 N FRIGATE AVE    CA  90744  33.777863 -118.278247   
5            5257 N NEWCASTLE AVE    CA  91316  34.166353 -118.524347   
6                   500 E 59TH PL    CA  90003  33.986083 -118.266182   
7             2409 N YORKSHIRE DR    CA  90065  34.110697 -118.230535   
9             3548 N MULTIVIEW DR    CA  90068  34.130287 -118.360437   
10          4758 N SUNNYSLOPE AVE    CA  91423  34.157384 -118.426876   
11      901 N STRADA VECCHIA ROAD    CA  90077  34.089718 -118.443840   
13             5869 W DAUPHIN AVE    CA  90034  27.871628  -82.498600   
14            6951 N ETIWANDA AVE    CA  91335  34.197043 -118.531672   
17           213 N TIGERTAIL ROAD    CA  90049  34.061779 -118.479050   
18                1021 W LOWEN ST    CA  90744  33.795173 -118.275929   
19             2038 S MALCOLM AVE    CA  90025  34.046825 -118.431553   
20           15935 W COMMUNITY ST    CA  91343  34.222889 -118.479941   
22               1027 S CENTRE ST    CA  90731  33.734600 -118.283549   
24              13640 W GARBER ST    CA  91331  34.248274 -118.430194   
25              5211 N DARRO ROAD    CA  91364  34.163874 -118.599916   
28             929 S MUIRFIELD RD    CA  90019  34.058004 -118.330674   
30              7217 S DENKER AVE    CA  90047  33.973995 -118.304625   
31             12383 W COVELLO ST    CA  91605  34.208036 -118.402962   
34                 1334 E 57TH ST    CA  90011  33.991109 -118.251275   
36             9470 S HOBART BLVD    CA  90047  33.949010 -118.307858   
37            2568 S MILITARY AVE    CA  90064  34.035105 -118.430608   
40                 1871 W 25TH ST    CA  90018  34.033650 -118.305059   
41            2651 N ABERDEEN AVE    CA  90027  34.116947 -118.289847   
42             7129 W HILLROSE ST    CA  91042  34.262797 -118.286691   
43               19501 W ROMAR ST    CA  91324  34.254537 -118.558210   
..                            ...   ...    ...        ...         ...   
60                  831 E 94TH ST    CA  90002  33.951132 -118.259707   
62             14703 S DENVER AVE    CA  90248  33.898751 -118.284438   
63       5243 N NORTH MAYWOOD AVE    CA  90041  34.141743 -118.211699   
64         10000 N OWENSMOUTH AVE    CA  91311  34.252303 -118.601664   
65             2415 S GLYNDON AVE    CA  90291  33.995280 -118.448793   
66               4510 N STARK AVE    CA  91364  34.151990 -118.623749   
67               1713 N QUEENS CT    CA  90069  34.101299 -118.377800   
68         12006 N HIGHWATER ROAD    CA  91344  34.289339 -118.519211   
69                6701 N SALE AVE    CA  91307  34.192505 -118.618545   
72                 1642 W 22ND ST    CA  90732  33.725276 -118.312858   
73          14333 W VAN NUYS BLVD    CA  91331  34.248674 -118.445968   
74            10650 W CHIQUITA ST    CA  91602  34.144120 -118.364003   
75             13463 W HUBBARD ST    CA  91342  34.310668 -118.426682   
76               2758 N DENBY AVE    CA  90039  34.102177 -118.245930   
78             2412 N EASTERN AVE    CA  90032  34.073762 -118.180319   
80               12818 W FRIAR ST    CA  91606  34.185637 -118.412549   
81          16058 W LOS ALIMOS ST    CA  91344  34.265526 -118.482304   
82                2425 S CLOY AVE    CA  90291  33.985706 -118.459419   
83            2301 S LA SALLE AVE    CA  90018  34.035432 -118.305777   
84           18029 W BURBANK BLVD    CA  91316  34.172216 -118.525399   
85            806 N MANSFIELD AVE    CA  90038  34.085567 -118.340628   
88               18215 W RAYEN ST    CA  91325  34.231862 -118.530020   
89                 1224 E 57TH ST    CA  90011  33.991109 -118.253724   
90  9113 #3